File: EmployeeManagementSystem/src/main/java/org/mahesh/ems/service/AttendanceService.java
package org.mahesh.ems.service;

import org.mahesh.ems.dto.AttendanceResponse;

import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {

    AttendanceResponse checkIn(String email);

    AttendanceResponse checkOut(String email);

    List<AttendanceResponse> getMyAttendance(String email);

    List<AttendanceResponse> getAllAttendance();

    List<AttendanceResponse> getAttendanceByDate(LocalDate date);

}

File: ems-frontend/src/services/attendanceService.js
import API from "../api/axiosConfig";

export const checkIn = () => {
    return API.post("/attendance/check-in");
};

export const checkOut = () => {
    return API.put("/attendance/check-out");
};

export const getMyAttendance = () => {
    return API.get("/attendance/me");
};

export const getAllAttendance = () => {
    return API.get("/attendance");
};

export const getAttendanceByDate = (date) => {
    return API.get(`/attendance/date?date=${date}`);
};